ui/widgets/datepicker.js

```diff
@@ -1036,7 +1036,7 @@ $.extend( Datepicker.prototype, {
 		}
 
 		inst = this._getInst( target[ 0 ] );
-		inst.selectedDay = inst.currentDay = $( "a", td ).html();
+		inst.selectedDay = inst.currentDay = $( "a", td ).data( "date" );
```

Comment:
Actually, now that I look at it again, we should read the attribute directly instead of using `data`. `.data(key)` pulls the initial value from the `data-key` attribute but that's it; it then keeps its state and disregards the attribute. It sounds we want the attribute to be the soruce of truth here.

Most other similar places in this widget use `+this.getAttribute( "data-something" )`. I'm not sure if skipping jQuery's `.attr()` had any reason but we should either use `getAttribute` or `attr` here & cast the value to number.